stdlib/Makefile

```diff
@@ -0,0 +1,24 @@
+default: Pkg
+SRCDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
+JULIAHOME := $(abspath $(SRCDIR)/..)
+SRCCACHE := $(abspath $(SRCDIR)/srccache)
+BUILDDIR := $(SRCCACHE)
+
+include $(JULIAHOME)/Make.inc
+include $(JULIAHOME)/deps/tools/common.mk
+include $(JULIAHOME)/deps/tools/git-external.mk
+
+PKG := https://github.com/JuliaLang/Pkg.jl
+PKG_TAR_URL = https://api.github.com/repos/JuliaLang/Pkg.jl/tarball/$1
+$(eval $(call git-external,Pkg,PKG,,,$(BUILDDIR)))
+
+extract-pkg: $(BUILDDIR)/$(PKG_SRC_DIR)/source-extracted
+Pkg: $(BUILDDIR)/$(PKG_SRC_DIR)/source-extracted
```

Comment:
Shouldn't this check for `Pkg/source-extracted` rather than the file in the build location?

Reply:
> `$(call git-external,Pkg,PKG,,,$(BUILDDIR))`

This tells `git-external` to download and extract it into `$(BUILDDIR)`.  See [this line](https://github.com/JuliaLang/julia/blob/d64ccdb7c1d1fb3370be1c0c1b96bd8bb6f5fc12/deps/tools/git-external.mk#L62).